src/app/features/users-page/users-page.component.ts

```diff
@@ -1,24 +1,63 @@
-import { Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
 import { IUsers } from '@features/users-page/interfaces/users';
 import { getUsersPending } from '@features/users-page/store/users.actions';
 import { getUsersFromState } from '@features/users-page/store/users.selectors';
 import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { CompareFilters } from '@features/users-page/filter-bar/interfaces/compare-filters';
+
+function compareUserFilter (user: IUsers, { type, value }: CompareFilters) {
```

Comment:
no local functions, make it private method or move to a different file
+ strange name, if you will translate it will be "сравнить пользователя фильтровать"
just name `filterUser`